lib/detect-testing-library-utils.ts

```diff
@@ -308,55 +372,38 @@ export function detectTestingLibraryUtils<
     };
 
     /**
-     * Determines whether a given node is fireEvent method or not
+     * Determines whether a given node is fireEvent util itself or not.
+     *
+     * Not to be confused with {@link isFireEventMethod}
      */
-    const isFireEventMethod: IsFireEventMethodFn = (node) => {
-      const fireEventUtil = findImportedUtilSpecifier(FIRE_EVENT_NAME);
-      let fireEventUtilName: string | undefined;
-
-      if (fireEventUtil) {
-        fireEventUtilName = ASTUtils.isIdentifier(fireEventUtil)
-          ? fireEventUtil.name
-          : fireEventUtil.local.name;
-      } else if (isAggressiveModuleReportingEnabled()) {
-        fireEventUtilName = FIRE_EVENT_NAME;
-      }
-
-      if (!fireEventUtilName) {
-        return false;
-      }
-
-      const parentMemberExpression:
-        | TSESTree.MemberExpression
-        | undefined = isMemberExpression(node.parent) ? node.parent : undefined;
-
-      if (!parentMemberExpression) {
-        return false;
-      }
-
-      // make sure that given node it's not fireEvent object itself
-      if (
-        [fireEventUtilName, FIRE_EVENT_NAME].includes(node.name) ||
-        (ASTUtils.isIdentifier(parentMemberExpression.object) &&
-          parentMemberExpression.object.name === node.name)
-      ) {
-        return false;
-      }
-
-      // check fireEvent.click() usage
-      const regularCall =
-        ASTUtils.isIdentifier(parentMemberExpression.object) &&
-        parentMemberExpression.object.name === fireEventUtilName;
+    const isFireEventUtil = (node: TSESTree.Identifier): boolean => {
+      return isTestingLibraryUtil(
+        node,
+        (identifierNodeName, originalNodeName) => {
+          return [identifierNodeName, originalNodeName].includes('fireEvent');
+        }
+      );
+    };
 
-      // check testingLibraryUtils.fireEvent.click() usage
-      const wildcardCall =
-        isMemberExpression(parentMemberExpression.object) &&
-        ASTUtils.isIdentifier(parentMemberExpression.object.object) &&
-        parentMemberExpression.object.object.name === fireEventUtilName &&
-        ASTUtils.isIdentifier(parentMemberExpression.object.property) &&
-        parentMemberExpression.object.property.name === FIRE_EVENT_NAME;
+    /**
+     * Determines whether a given node is userEvent util itself or not.
+     *
+     * Not to be confused with {@link isUserEventMethod}
```

Comment:
`isUserEventMethod` doesn't exist yet, but as I said it's coming in the next PR.